task.sh

```diff
@@ -1 +1,42 @@
 #! /bin/bash
+
+BACKUP_PATH="ShopDB_backup.sql"
+DATA_BACKUP_PATH="ShopDB_data_backup.sql"
+
+if [ -z "$DB_USER" ] || [ -z "$DB_PASSWORD" ]; then
+  echo "Error: Environment DB_USER or DB_PASSWORD do not exist."
+  exit 1
+fi
+
+check_error() {
+  if [ $? -ne 0 ]; then
+    echo "Error: $1"
+    exit 1
+  fi
+}
+
+echo "Creating full backup of ShopDB..."
+mysqldump -u "$DB_USER" -p"$DB_PASSWORD" \
+  --databases ShopDB --result-file="$BACKUP_PATH" --no-create-db
+check_error "Failed to create full backup of ShopDB."
+
+echo "Restoring backup to ShopDBReserve..."
+mysql -u "$DB_USER" -p"$DB_PASSWORD" \
```

Comment:
Consider adding a check to ensure that the `ShopDBReserve` database does not contain any important data before dropping it. This can prevent accidental data loss.